### DISCUSSION TITLE
Translation Helper Script - Adding missing translation keys

### DISCUSSION THREAD

**Discussion by Spokeek** [vrc-get/vrc-get#1495] [Ideas]
A small proposal for a helping script i made and used for my translation needs in french.

I made this discussion after already making the script, but it's also if you want to try it and see if it could suit your needs.

https://github.com/Spokeek/vrc-get/blob/french-translation/.github/scripts/load-missing-localization-keys.js

**Comment by Spokeek** [comment]
adding your discussions here.

It was mentioned that the keys are sorted by category, but this is also done by the key prefix, so IMO this order is kept while sorting alphabetically automatically. I'm not against adding more style to increase human readability of course.

Also, to confirm if i'm not wrong @anatawa12 the format we follow is JSON5 and not JSON. so i know if the format that is done here is fine or not.

And @JustBuddy , Just so i know and feel free to express it if that's your opinion, but is just the idea of such script something you are disagreeing with ?
So i know if simply i should keep that on the side or if this can interest you (for now you're the only one giving interest after all 👍

**Reply by JustBuddy** [reply]
@Spokeek 
Don't misunderstand me, I am not against scripts if they solve an actual problem and help make maintainers do quicker work.
For me, as I am rather quick with submitting translations, I am certainly faster doing it manually in my text editor, just comparing diffs if needed. Mostly I just search the newly added key in the English main file and add it to the same line on my German translation, re-sorting if I notice any difference in on which line keys are.

It's faster for me and preserves the good readability we have thanks to @Sayamame-beans restructuring.

**Reply by Spokeek** [reply]
Gotcha, no feelings at all, it's all in good will and just wanted to know where the discussion was 👍

**Reply by anatawa12** [reply]
I'm using JSON5 only for commenting and trailing commas so I like to keep them similar to JSON.

**Reply by Spokeek** [reply]
Gotcha so based on that, i don't really see a way to do that automatically, at least without having pain making a script for that.
At least for now from the feedback i'll remove the script and add my script in a gist like Sayamame did.

I'll think this further and if a need is expressed, i'll pull it out.
For the rest thank you for the feedback.

**Comment by Sayamame-beans** [comment]
[off-topic]
I had forgotten to publish my helper script I used for reordering keys. I published it.
- #1511

**Comment by Spokeek** [comment]
Updated, added my gist following the idea of Sayamame.

https://gist.github.com/Spokeek/f7e478d7c293f04c295943252ca7f941

I'm closing this discussion for now. Feel free to reopen if you have something to add.
